pkg/jsonnet/implementation/binary/impl.go

```diff
@@ -0,0 +1,51 @@
+package binary
+
+import (
+	"os/exec"
+	"strconv"
+
+	"github.com/grafana/tanka/pkg/jsonnet/implementation/types"
+)
+
+type JsonnetBinaryRunner struct {
```

Comment:
It'd be nice to collect statistics about how long we spend in the external binary (number of times we execute, time spent in each execution, total time spent in all of them) and output that. Two reasons:

  1. Forking off processes isn't free and if we do it a lot it could add up. With a bit more instrumentation on the native codepaths (could be done later) we could do a like-for-like comparison with go-jsonnet and `jsonnet` the binary which would give interesting detail on the overhead added by this.
  2. Gives interesting comparisons on the speed of the implementations themselves, it's a nice boundary between Tanka's code and the Jsonnet implementation's.

Reply:
It's nice to have more detailed logs to know that with certainty. 99% feels like a good number but you would actually have that data for sure if you had trace logs (or proper trace spans). Maybe we'd get surprised. Or when using jrsonnet you could see that now Tanka's percentage is actually higher and that makes the Tanka team work harder on optimising.

But fine, I don't insist. If you feel very strongly then sure - it's not _required_. :slightly_smiling_face:

Reply:
Yeah, I'll do https://github.com/grafana/tanka/pull/907 rather than logs with runtimes